numpyro/distributions/truncated.py

```diff
@@ -67,11 +67,15 @@ def sample(self, key, sample_shape=()):
         finfo = jnp.finfo(dtype)
         minval = finfo.tiny
         u = random.uniform(key, shape=sample_shape + self.batch_shape, minval=minval)
+        return self.icdf(u)
+
+    def icdf(self, q):
         loc = self.base_dist.loc
         sign = jnp.where(loc >= self.low, 1.0, -1.0)
-        return (1 - sign) * loc + sign * self.base_dist.icdf(
-            (1 - u) * self._tail_prob_at_low + u * self._tail_prob_at_high
+        ppf = (1 - sign) * loc + sign * self.base_dist.icdf(
+            (1 - q) * self._tail_prob_at_low + q * self._tail_prob_at_high
         )
+        return jnp.where(q < 0, jnp.nan, ppf)
```

Comment:
why this where?

Reply:
Please see my above comment. If this is not implemented, the icdf value will return a numeric, non NaN value outside the input range of [0,1]. The behavior of other NumPyro distributions is to return NaN outside this range. A non-NaN value outside of the input range can potentially cause bugs and/or confusion. Of course in principle one shouldn’t call the icdf with a value outside this input range, but I figured it would be best practice to ensure expected behavior.

Reply:
I could also imagine doing the following: If we're worried about the performance hit to `sample` because of the added `jnp.where` call, then we could wrap the un-masked `icdf` code in a separate helper function and only mask out-of-range values for the `icdf` method. If it's preferred to keep the behaviour where numeric values are returned when out of bounds, then I'm happy to revert the previous commit :-)

Reply:
I think having the check is fine. Just curious, why not checking both q < 0 and q > 1?

Reply:
If it’s a left/right truncated distribution, we only need to check on the relevant side. Otherwise the base distribution icdf handles if it’s out of range.

Reply:
Thanks for explaining!!